packages/asset-swapper/src/utils/market_operation_utils/bancor_service.ts

```diff
@@ -1,65 +1,34 @@
+import { BlockchainType } from '@bancor/sdk/dist/types';
 
-import { BancorFillData, Quote } from './types';
+import { TOKENS } from './constants';
 
-/**
- * Converts an address to a Bancor Token type
- */
-export function token(address: string, blockchainType: BlockchainType = BlockchainType.Ethereum): Token {
-    return {
-        blockchainType,
-        blockchainId: address,
-    };
-}
+const findToken = (tokenAddress: string, graph: object): string =>
+    // If we're looking for WETH it is stored by Bancor as the 0xeee address
+    tokenAddress.toLowerCase() === TOKENS.WETH.toLowerCase()
+        ? '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE'
+        : Object.keys(graph).filter(k => k.toLowerCase() === tokenAddress.toLowerCase())[0];
 
 export class BancorService {
-    // Bancor recommends setting this value to 2% under the expected return amount
-    public minReturnAmountBufferPercentage = 0.99;
-
     public static async createAsync(provider: SupportedProvider): Promise<BancorService> {
         const sdk = await SDK.create({ ethereumNodeEndpoint: provider });
         const service = new BancorService(sdk);
         return service;
     }
 
     constructor(public sdk: SDK) {}
-
-    public async getQuotesAsync(
-        fromToken: string,
-        toToken: string,
-        amounts: BigNumber[],
-    ): Promise<Array<Quote<BancorFillData>>> {
-        const sdk = this.sdk;
-        const blockchain = sdk._core.blockchains[BlockchainType.Ethereum] as Ethereum;
-        const sourceDecimals = await getDecimals(blockchain, fromToken);
-        const quotes = await sdk.pricing.getPathAndRates(
-            token(fromToken),
-            token(toToken),
-            amounts.map(amt => fromWei(amt.toString(), sourceDecimals)),
-        );
-        const targetDecimals = await getDecimals(blockchain, toToken);
-        const networkAddress = this.getBancorNetworkAddress();
-
-        return quotes.map(quote => {
-            const { path, rate } = quote;
-            const output = toWei(rate, targetDecimals);
-            return {
-                amount: new BigNumber(output).multipliedBy(this.minReturnAmountBufferPercentage).dp(0),
-                fillData: {
-                    path: path.map(p => p.blockchainId),
-                    networkAddress,
-                },
-            };
-        });
-    }
-
-    public getBancorNetworkAddress(): string {
-        const blockchain = this.sdk._core.blockchains[BlockchainType.Ethereum] as Ethereum;
-        return blockchain.bancorNetwork._address;
+    public getPaths(_fromToken: string, _toToken: string): string[][] {
+        // HACK: We reach into the blockchain object and pull in it's cache of tokens
+        // and we use it's internal non-async getPathsFunc
+        try {
+            const blockchain = this.sdk._core.blockchains[BlockchainType.Ethereum] as Ethereum;
+            const fromToken = findToken(_fromToken, blockchain.graph);
```

Comment:
Does this graph just keep itself updated continuously or does it need to be triggered?

Reply:
there's a background `refresh` running in the sdk